sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs

```diff
@@ -240,16 +238,121 @@ public async Task EventHub_PartitionKey()
+            }
+        }
+
+        [Test]
+        public async Task EventHub_InitialOffsetFromEnd()
+        {
+            // Send a message to ensure the stream is not empty as we are trying to validate that no messages are delivered in this case
+            var producer = new EventHubProducerClient(EventHubsTestEnvironment.Instance.EventHubsConnectionString, _eventHubScope.EventHubName);
+            await producer.SendAsync(new EventData[] { new EventData(new BinaryData(_testId)) });
+
+            var (jobHost, host) = BuildHost<EventHubTestInitialOffsetFromStartEndJobs>(
+                builder =>
+                    builder.ConfigureServices(services =>
+                    {
+                        services.Configure<EventHubOptions>(options =>
+                        {
+                            options.InitialOffsetOptions.Type = "FromEnd";
+                            options.AddSender(_eventHubScope.EventHubName, EventHubsTestEnvironment.Instance.EventHubsConnectionString);
+                            options.AddReceiver(_eventHubScope.EventHubName, EventHubsTestEnvironment.Instance.EventHubsConnectionString);
+                        });
+                    }));
+            using (jobHost)
+            {
+                // We don't expect to get signalled as there should be no messages received with a FromEnd initial offset
+                bool result = _eventWait.WaitOne(Timeout);
+                Assert.False(result, "An event was received while none were expected.");
+
+                // send a new event which should be received
+                await producer.SendAsync(new EventData[] { new EventData(new BinaryData(_testId)) });
+                result = _eventWait.WaitOne(Timeout);
+                Assert.True(result);
+            }
+        }
+
+        [Test]
+        public async Task EventHub_InitialOffsetFromEnqueuedTime()
+        {
+            // Mark the time now and send a message which should be the only one that is picked up when we run the actual test host
+
+            var producer = new EventHubProducerClient(EventHubsTestEnvironment.Instance.EventHubsConnectionString, _eventHubScope.EventHubName);
+            for (int i = 0; i < 3; i++)
+            {
+                // send one at a time so they will have slightly different enqueued times
+                await producer.SendAsync(new EventData[] { new EventData(new BinaryData(_testId)) });
+                await Task.Delay(1000);
+            }
+            var consumer = new EventHubConsumerClient(
+                EventHubConsumerClient.DefaultConsumerGroupName,
+                EventHubsTestEnvironment.Instance.EventHubsConnectionString,
+                _eventHubScope.EventHubName);
+
+            var events = consumer.ReadEventsAsync();
+            _initialOffsetEnqueuedTimeUTC = DateTime.UtcNow;
+            int ct = 0;
+            await foreach (PartitionEvent evt in events)
+            {
+                if (++ct == 1)
```

Comment:
It's a complicated way to do first :) There is also https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core.TestFramework/src/TestAsyncEnumerableExtensions.cs#L11

Reply:
Hah, yeah this wasn't as bad when I wasn't taking the first one. I don't need the ct I guess.